src/junitParser.ts

```diff
@@ -70,18 +69,17 @@ async function parseFile(file: string, projectTokenDictionaryStrs: string[]) {
     const data: string = fs.readFileSync(file, 'utf8');
     const parser = new XMLParser({ allowBooleanAttributes: true, ignoreAttributes: false, attributeNamePrefix: '' });
     const report = parser.parse(data) as Partial<JUnitReport>;
-
-    return parseSuite(report, file, projectTokenDictionaryStrs);
+    const testSuites = castArray(report.testsuites?.testsuite || []);
+    return Promise.all(testSuites.map(async testSuite => parseSuite(file, projectTokenDictionaryStrs, testSuite)));
 }
 
-async function parseSuite(report: JUnitReport, fileName: string, projectTokenDictionaryStrs: string[]) {
-    const testsuite = report.testsuite || report.testsuites?.testsuite;
-    const result: InternalTestResult = { fileName, name: testsuite?.name, totalCount: 0, skipped: 0, failedEvaluating: parseInt(testsuite?.['failure-evaluating'] || '0') || 0, annotations: [] };
-    if (!testsuite?.testcase) {
+async function parseSuite(fileName: string, projectTokenDictionaryStrs: string[], testSuite: JUnitSuite) {
```

Comment:
can we keep `testSuite` as `testsuite` to avoid unrelated changes?

Reply:
NP